### PR TITLE
[vi] Deep learning tips and tricks

### PR DESCRIPTION
**Pull request by tuanhm2911**
Hi,
I have finished my translation to Vietnamese in topic Deep Learning tips. 
Waiting for reviews.

### CODE REVIEW

**Comment by shervinea** [comment]
Great job @tuanhm2911, thank you for all your time! Let's start the review process now.

File: vi/cs-230-deep-learning-tips-and-tricks.md [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+**14. [Color shift, Noise addition, Information loss, Contrast change]**
+
+&#10230; [Dịch chuyển màu, Thêm nhiễu, Mất mát thông tin, Thay đổi độ tương phản]
+
+<br>
+
+
+**15. [Nuances of RGB is slightly changed, Captures noise that can occur with light exposure, Addition of noise, More tolerance to quality variation of inputs, Parts of image ignored, Mimics potential loss of parts of image, Luminosity changes, Controls difference in exposition due to time of day]**
+
+&#10230; [Các sắc thái của RGB bị thay đổi một chút, Nhiễu nhiễu có thể xảy ra khi tiếp xúc với ánh sáng nhẹ, Bổ sung nhiễu, Chịu được sự thay đổi chất lượng của các yếu tố đầu vào, Các phần của hình ảnh bị bỏ qua, Bắt chước mất khả năng của các phần của hình ảnh, Thay đổi độ sáng, Kiểm soát sự khác biệt do phơi sáng do thời gian trong ngày]
```
Comment (tuananhhedspibk) [inline vi/cs-230-deep-learning-tips-and-tricks.md]:
```suggestion
&#10230; [Các sắc thái của RGB bị thay đổi một chút, Captures noise có thể xảy ra khi tiếp xúc với ánh sáng nhẹ, Bổ sung nhiễu, Chịu được sự thay đổi chất lượng của các yếu tố đầu vào, Các phần của hình ảnh bị bỏ qua, Mô phỏng khả năng mất của các phần trong hình ảnh, Thay đổi độ sáng, Kiểm soát sự khác biệt do phơi sáng theo thời gian trong ngày]
```

File: vi/cs-230-deep-learning-tips-and-tricks.md [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+**42. [Momentum, Dampens oscillations, Improvement to SGD, 2 parameters to tune]**
+
+&#10230; [Momentum, Làm giảm dao động, Cải thiện SGD, 2 tham số để tinh chỉnh]
+
+<br>
+
+
+**43. [RMSprop, Root Mean Square propagation, Speeds up learning algorithm by controlling oscillations]**
+
+&#10230; [RMSprop, lan truyền Root Mean Square, Thuật toán tăng tốc bằng kiểm soát dao động]
```
Comment (tuananhhedspibk) [inline vi/cs-230-deep-learning-tips-and-tricks.md]:
```suggestion
&#10230; [RMSprop, lan truyền Root Mean Square, Thuật toán tăng tốc độ học bằng kiểm soát dao động]
```

File: vi/cs-230-deep-learning-tips-and-tricks.md [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+**40. Adaptive learning rates ― Letting the learning rate vary when training a model can reduce the training time and improve the numerical optimal solution. While Adam optimizer is the most commonly used technique, others can also be useful. They are summed up in the table below:**
+
+&#10230; Tốc độ học thích nghi - Để tốc độ học thay đổi khi huấn luyện một mô hình có thể giảm thời gian huấn luyện và cải thiện giải pháp tối ưu số. Trong khi Adam tối ưu hóa là kỹ thuật được sử dụng phổ biến nhất, những phương pháp khác cũng có thể hữu ích. Chúng được tóm tắt trong bảng dưới đây:
+
+<br>
+
+
+**41. [Method, Explanation, Update of w, Update of b]**
+
+&#10230; [Phương pháp, Giải thích, Cập nhật của w, Cập nhật của b]
```
Comment (tuananhhedspibk) [inline vi/cs-230-deep-learning-tips-and-tricks.md]:
```suggestion
&#10230; [Phương thức, Giải thích, Cập nhật của w, Cập nhật của b]
```

File: vi/cs-230-deep-learning-tips-and-tricks.md [hunk 4]
```diff
@@ -0,0 +1,456 @@
...
+**39. Learning rate ― The learning rate, often noted α or sometimes η, indicates at which pace the weights get updated. It can be fixed or adaptively changed. The current most popular method is called Adam, which is a method that adapts the learning rate.**
+
+&#10230; Tốc độ học - Tốc độ học, thường được kí hiệu là α hoặc đôi khi là η, cho biết mức độ thay đổi của các trọng số sau mỗi lần cập nhật. Nó có thể được cố định hoặc thay đổi thích ứng. Phương pháp phổ biến nhất hiện nay được gọi là Adam, đây là phương pháp thích nghi với tốc độ học.
+
+<br>
+
+
+**40. Adaptive learning rates ― Letting the learning rate vary when training a model can reduce the training time and improve the numerical optimal solution. While Adam optimizer is the most commonly used technique, others can also be useful. They are summed up in the table below:**
+
+&#10230; Tốc độ học thích nghi - Để tốc độ học thay đổi khi huấn luyện một mô hình có thể giảm thời gian huấn luyện và cải thiện giải pháp tối ưu số. Trong khi Adam tối ưu hóa là kỹ thuật được sử dụng phổ biến nhất, những phương pháp khác cũng có thể hữu ích. Chúng được tóm tắt trong bảng dưới đây:
```
Comment (tuananhhedspibk) [inline vi/cs-230-deep-learning-tips-and-tricks.md]:
```suggestion
&#10230; Tốc độ học thích nghi - Để cho tốc độ học thay đổi khi huấn luyện một mô hình có thể giảm thời gian huấn luyện và cải thiện giải pháp tối ưu số. Trong khi tối ưu hóa Adam (Adam optimizer) là kỹ thuật được sử dụng phổ biến nhất, nhưng những phương pháp khác cũng có thể hữu ích. Chúng được tổng kết trong bảng dưới đây:
```

File: vi/cs-230-deep-learning-tips-and-tricks.md [hunk 5]
```diff
@@ -0,0 +1,456 @@
...
+**38. Optimizing convergence**
+
+&#10230; Tối ưu hội tự
+
+<br>
+
+
+**39. Learning rate ― The learning rate, often noted α or sometimes η, indicates at which pace the weights get updated. It can be fixed or adaptively changed. The current most popular method is called Adam, which is a method that adapts the learning rate.**
+
+&#10230; Tốc độ học - Tốc độ học, thường được kí hiệu là α hoặc đôi khi là η, cho biết mức độ thay đổi của các trọng số sau mỗi lần cập nhật. Nó có thể được cố định hoặc thay đổi thích ứng. Phương pháp phổ biến nhất hiện nay được gọi là Adam, đây là phương pháp thích nghi với tốc độ học.
```
Comment (tuananhhedspibk) [inline vi/cs-230-deep-learning-tips-and-tricks.md]:
```suggestion
&#10230; Tốc độ học - Tốc độ học, thường được kí hiệu là α hoặc đôi khi là η, cho biết mức độ thay đổi của các trọng số sau mỗi lần được cập nhật. Nó có thể được cố định hoặc thay đổi thích ứng. Phương thức phổ biến nhất hiện nay là Adam, đây là phương thức thích nghi với tốc độ học.
```

File: vi/cs-230-deep-learning-tips-and-tricks.md [hunk 6]
```diff
@@ -0,0 +1,456 @@
...
+**37. [Freezes all layers, trains weights on softmax, Freezes most layers, trains weights on last layers and softmax, Trains weights on layers and softmax by initializing weights on pre-trained ones]**
+
+&#10230; [Đông đặc các lớp, huấn luyện hàm softmax, Đông đặc hầu hết các lớp, huấn luyện trên lớp cuối và hàm softmax, Huấn luyện trọng số trên tầng và softmax với khởi tạo trọng số trên mô hình đã huấn luyện sẵn]
+
+<br>
+
+
+**38. Optimizing convergence**
+
+&#10230; Tối ưu hội tự
```
Comment (tuananhhedspibk) [inline vi/cs-230-deep-learning-tips-and-tricks.md]:
```suggestion
&#10230; Tối ưu hội tụ
```

File: vi/cs-230-deep-learning-tips-and-tricks.md [hunk 7]
```diff
@@ -0,0 +1,456 @@
...
+**36. [Small, Medium, Large]**
+
+&#10230; [Nhỏ, Trung bình, Lớn]
+
+<br>
+
+
+**37. [Freezes all layers, trains weights on softmax, Freezes most layers, trains weights on last layers and softmax, Trains weights on layers and softmax by initializing weights on pre-trained ones]**
+
+&#10230; [Đông đặc các lớp, huấn luyện hàm softmax, Đông đặc hầu hết các lớp, huấn luyện trên lớp cuối và hàm softmax, Huấn luyện trọng số trên tầng và softmax với khởi tạo trọng số trên mô hình đã huấn luyện sẵn]
```
Comment (tuananhhedspibk) [inline vi/cs-230-deep-learning-tips-and-tricks.md]:
```suggestion
&#10230; [Cố định các tầng, huấn luyện trọng số trên hàm softmax, Cố định hầu hết các tầng, huấn luyện trọng số trên tầng cuối và hàm softmax, Huấn luyện trọng số trên tầng và softmax bằng việc khởi tạo trọng số trên mô hình đã huấn luyện sẵn]
```

File: vi/cs-230-deep-learning-tips-and-tricks.md [hunk 8]
```diff
@@ -0,0 +1,456 @@
...
+**33. Xavier initialization ― Instead of initializing the weights in a purely random manner, Xavier initialization enables to have initial weights that take into account characteristics that are unique to the architecture.**
+
+&#10230; Khởi tạo Xavier - Thay vì khởi tạo trọng số một cách ngẫu nhiên, khởi tạo Xavier cho chúng ta một cách khởi tạo tham số dựa trên một đặc tính độc nhất của mô hình.
+
+<br>
+
+
+**34. Transfer learning ― Training a deep learning model requires a lot of data and more importantly a lot of time. It is often useful to take advantage of pre-trained weights on huge datasets that took days/weeks to train, and leverage it towards our use case. Depending on how much data we have at hand, here are the different ways to leverage this:**
+
+&#10230; Học chuyển tiếp - Huấn luyện một mô hình học tập sâu đòi hỏi nhiều dữ liệu và quan trọng hơn là rất nhiều thời gian. Sẽ rất hữu ích để tận dụng các trọng số được đào tạo trước trên các bộ dữ liệu khổng lồ mất vài ngày / tuần để đào tạo và tận dụng nó cho trường hợp sử dụng của chúng ta. Tùy thuộc vào lượng dữ liệu chúng ta có trong tay, đây là các cách khác nhau để tận dụng điều này:
```
Comment (tuananhhedspibk) [inline vi/cs-230-deep-learning-tips-and-tricks.md]:
```suggestion
&#10230; Transfer learning - Huấn luyện một mô hình deep learning đòi hỏi nhiều dữ liệu và quan trọng hơn là rất nhiều thời gian. Sẽ rất hữu ích để tận dụng các trọng số đã được huyến luyện trước trên các bộ dữ liệu rất lớn mất vài ngày / tuần để huấn luyện và tận dụng nó cho trường hợp (use case) của chúng ta. Tùy thuộc vào lượng dữ liệu chúng ta có trong tay, đây là các cách khác nhau để tận dụng điều này:
```

File: vi/cs-230-deep-learning-tips-and-tricks.md [hunk 9]
```diff
@@ -0,0 +1,456 @@
...
+**32. Weights initialization**
+
+&#10230; Khởi tạo trọng số
+
+<br>
+
+
+**33. Xavier initialization ― Instead of initializing the weights in a purely random manner, Xavier initialization enables to have initial weights that take into account characteristics that are unique to the architecture.**
+
+&#10230; Khởi tạo Xavier - Thay vì khởi tạo trọng số một cách ngẫu nhiên, khởi tạo Xavier cho chúng ta một cách khởi tạo tham số dựa trên một đặc tính độc nhất của mô hình.
```
Comment (tuananhhedspibk) [inline vi/cs-230-deep-learning-tips-and-tricks.md]:
```suggestion
&#10230; Khởi tạo Xavier - Thay vì khởi tạo trọng số một cách ngẫu nhiên, khởi tạo Xavier cho chúng ta một cách khởi tạo trọng số dựa trên một đặc tính độc nhất của kiến trúc mô hình.
```

File: vi/cs-230-deep-learning-tips-and-tricks.md [hunk 10]
```diff
@@ -0,0 +1,456 @@
...
+**29. [Step 1: Take a batch of training data and perform forward propagation to compute the loss, Step 2: Backpropagate the loss to get the gradient of the loss with respect to each weight, Step 3: Use the gradients to update the weights of the network.]**
+
+&#10230; [Bước 1: Lấy một loạt dữ liệu huấn luyện và thực hiện lan truyền thẳng để tính toán mất mát, Bước 2: Sao lưu lại mất mát để có được độ dốc của mất mát theo từng trọng số, Bước 3: Sử dụng độ dốc để cập nhật trọng số của mạng.]
+
+<br>
+
+
+**30. [Forward propagation, Backpropagation, Weights update]**
+
+&#10230; [Lan truyền thẳng, Lan truyền ngược, Cập nhật trọng số]
```
Comment (tuananhhedspibk) [inline vi/cs-230-deep-learning-tips-and-tricks.md]:
```suggestion
&#10230; [Lan truyền xuôi, Lan truyền ngược, Cập nhật trọng số]
```

**Comment by tuananhhedspibk** [comment]
@tuanhm2911 Finished review for Deep learning tips and tricks in Vietnamese translation.

File: vi/cs-230-deep-learning-tips-and-tricks.md [hunk 11]
```diff
@@ -0,0 +1,456 @@
+**Deep Learning Tips and Tricks translation** [[webpage]](https://stanford.edu/~shervine/teaching/cs-230/cheatsheet-deep-learning-tips-and-tricks)
+
+<br>
+
+**1. Deep Learning Tips and Tricks cheatsheet**
+
+&#10230; Một số mẹo trong học sâu cheatsheet
```
[the remark this answers is not in the document]
Comment (damminhtien) [reply vi/cs-230-deep-learning-tips-and-tricks.md]:
```suggestion
&#10230; Cheatsheet về một số thủ thuật trong Deep Learning
```

File: vi/cs-230-deep-learning-tips-and-tricks.md [hunk 12]
```diff
@@ -0,0 +1,456 @@
...
+**6. [Parameter tuning, Xavier initialization, Transfer learning, Learning rate, Adaptive learning rates]**
+
+&#10230; [Tinh chỉnh tham số, Khởi tạo Xavier, Học chuyển tiếp, Tốc độ học, Tốc độ học đáp ứng]
+
+<br>
+
+
+**7. [Regularization, Dropout, Weight regularization, Early stopping]**
+
+&#10230; [Sự phạt mô hình, Dropout, Khởi tạo trọng số, Dừng sớm]
```
[the remark this answers is not in the document]
Comment (damminhtien) [reply vi/cs-230-deep-learning-tips-and-tricks.md]: I totally agree with @tuananhhedspibk that keep some terminologies in english for more clear meaning.
```suggestion
&#10230; [Regularization, Dropout, Weight regularization, Kỹ thuật Dừng sớm]
```

File: vi/cs-230-deep-learning-tips-and-tricks.md [hunk 13]
```diff
@@ -0,0 +1,456 @@
...
+**46. Regularization**
+
+&#10230; Phạt mô hình
+
+<br>
+
+
+**47. Dropout ― Dropout is a technique used in neural networks to prevent overfitting the training data by dropping out neurons with probability p>0. It forces the model to avoid relying too much on particular sets of features.**
+
+&#10230; Dropout - Dropout là một kỹ thuật được sử dụng trong các mạng nhân tạo để ngăn chặn hiện tượng quá khớp bằng cách loại bỏ các nơ-ron với xác suất p>0. Nó buộc mô hình tránh phụ thuộc quá nhiều vào một tập thuộc tính nào đó.
```
Comment (damminhtien) [inline vi/cs-230-deep-learning-tips-and-tricks.md]:
```suggestion
&#10230; Dropout - Dropout là một kỹ thuật được sử dụng trong các mạng nơ-ron để tránh overfitting trên tập huấn luyện bằng cách bỏ qua các nơ-ron với xác suất p>0. Nó giúp mô hình không bị phụ thuộc quá nhiều vào một tập thuộc tính nào đó.
```

**Comment by shervinea** [comment]
Hi @tuanhm2911, would the suggestions of @damminhtien and @tuananhhedspibk sound good to you? Please feel to give your feedback when you get the time. After that, we'll be able merge all your hard work!

File: vi/cs-230-deep-learning-tips-and-tricks.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (tuanhm2911) [reply vi/cs-230-deep-learning-tips-and-tricks.md]: thank you fir your suggestion

File: vi/cs-230-deep-learning-tips-and-tricks.md [hunk 14]
```diff
@@ -0,0 +1,456 @@
...
+**3. Tips and tricks**
+
+&#10230; Mẹo và thủ thuật
+
+<br>
+
+
+**4. [Data processing, Data augmentation, Batch normalization]**
+
+&#10230; [Xử lí dữ liệu, Thêm dữ liệu, Chuẩn hóa batch]
```
[the remark this answers is not in the document]
Comment (tuanhm2911) [reply vi/cs-230-deep-learning-tips-and-tricks.md]: thank you for your suggestion

File: vi/cs-230-deep-learning-tips-and-tricks.md [hunk 15]
```diff
@@ -0,0 +1,456 @@
...
+
+<br>
+
+
+**5. [Training a neural network, Epoch, Mini-batch, Cross-entropy loss, Backpropagation, Gradient descent, Updating weights, Gradient checking]**
+
+&#10230; [Huấn luyện một mô hình nhân tạo, Epoch, Mini-batch, Cross-entropy loss, Lan truyền ngược, Gradient descent, Cập nhật trọng số, Kiểm tra gradient]
```
[the remark this answers is not in the document]
Comment (tuanhm2911) [reply vi/cs-230-deep-learning-tips-and-tricks.md]: thank you for your suggestion

**Comment by tuananhhedspibk** [comment]
I think it's OK to be merged.

**Comment by shervinea** [comment]
Thank you everyone for your work! Moving forward with the merge now.

**Pull request merged**